pylint/checkers/base/basic_checker.py

```diff
@@ -657,13 +657,38 @@ def _check_misplaced_format_function(self, call_node: nodes.Call) -> None:
             ):
                 self.add_message("misplaced-format-function", node=call_node)
 
+    @staticmethod
+    def _is_sys_exit(node: nodes.Call) -> bool:
+        """Detect call to sys.exit."""
+        if not isinstance(node.func, nodes.Attribute) or isinstance(
+            node.parent, nodes.Lambda
+        ):
+            return False
+
+        try:
+            inferred = next(node.func.infer())
```

Comment:
What happen if the call to next raise a ``StopIteration`` ?

Reply:
Why not use `safe_infer`?

Reply:
> Why not use `safe_infer`?

because I'm just hearing about it 🤣 

Joking aside, I took the example from the many other uses of this pattern in the repo
```
        try:
            inferred = list(node....infer())
        except astroid.InferenceError:
            return
```
so should these examples also be moved to use safe_infer which seems to do all the catching we need? (separate issue)

Reply:
Some... 

Sometimes we want to iterate over all inference results and just handle the exceptions. For example, for file wrappers the inference is flaky as the `io` module does things we don't understand. However, because we know this we don't want to focus on the flaky total result but focus on the first sub-result as we can determine whether we are looking at an `io` object from it. I know that that is not the best explanation, but I can't think of a better example right now.

That said, if you're looking to contribute checking all `infer()` calls and seeing whether their exception branched are covered would be a good way to start removing some of them. I think there is actually an issue in `astroid` about doing the same there.

_For a future project we could even look at something like `safe_iterating_infer` to catch some but not all exceptions and return an iterator, but that's not something we should look at right now_